homeassistant/components/upnp/sensor.py

```diff
@@ -80,17 +139,24 @@ async def async_setup_entry(
 
     LOGGER.debug("Adding sensors")
 
-    sensors = [
-        RawUpnpSensor(coordinator, SENSOR_TYPES[BYTES_RECEIVED]),
-        RawUpnpSensor(coordinator, SENSOR_TYPES[BYTES_SENT]),
-        RawUpnpSensor(coordinator, SENSOR_TYPES[PACKETS_RECEIVED]),
-        RawUpnpSensor(coordinator, SENSOR_TYPES[PACKETS_SENT]),
-        DerivedUpnpSensor(coordinator, SENSOR_TYPES[BYTES_RECEIVED]),
-        DerivedUpnpSensor(coordinator, SENSOR_TYPES[BYTES_SENT]),
-        DerivedUpnpSensor(coordinator, SENSOR_TYPES[PACKETS_RECEIVED]),
-        DerivedUpnpSensor(coordinator, SENSOR_TYPES[PACKETS_SENT]),
-    ]
-    async_add_entities(sensors)
+    entities = []
+    entities.append(
+        RawUpnpSensor(
+            coordinator=coordinator,
+            sensor_entity=sensor_entity,
+        )
+        for sensor_entity in SENSOR_ENTITY_DESCRIPTIONS[RAW_SENSOR]
+    )
+
+    entities.append(
+        DerivedUpnpSensor(
+            coordinator=coordinator,
+            sensor_entity=sensor_entity,
+        )
+        for sensor_entity in SENSOR_ENTITY_DESCRIPTIONS[DERIVED_SENSOR]
+    )
+
+    async_add_entities(entities)
 
 
 class UpnpSensor(UpnpEntity, SensorEntity):
```

Comment:
And this under here:

``` 
    coordinator: UpnpDataUpdateCoordinator
    entity_description: UpnpSensorEntityDescription
```

Reply:
Not 100% sure if I got what you meant, but I think so. :-)

Reply:
Understand now. :-)